crates/aranya-client-capi/src/api/defs.rs

```diff
@@ -1232,6 +1245,72 @@ pub unsafe fn team_roles(
     Ok(())
 }
 
+/// Create a role.
+pub fn create_role(
+    client: &Client,
+    team: &TeamId,
+    role_name: RoleName,
+    owning_role: &RoleId,
+    role_out: &mut MaybeUninit<Role>,
+) -> Result<(), imp::Error> {
+    // SAFETY: Caller must ensure `name` is a valid C String.
+    let role_name = unsafe { role_name.as_underlying() }?;
+    let role = client.rt.block_on(
+        client
+            .inner
+            .team(team.into())
+            .create_role(role_name, owning_role.into()),
+    )?;
+    Role::init(role_out, role);
+    Ok(())
+}
+
+/// Delete a role.
+pub fn delete_role(client: &Client, team: &TeamId, role: &RoleId) -> Result<(), imp::Error> {
+    client
+        .rt
+        .block_on(client.inner.team(team.into()).delete_role(role.into()))?;
+    Ok(())
+}
+
+/// Assign a permission to a role.
+pub fn add_perm_to_role(
+    client: &Client,
+    team: &TeamId,
+    role: &RoleId,
+    perm: Permission,
+) -> Result<(), imp::Error> {
+    // SAFETY: Caller must ensure `perm` is a valid C String.
+    let perm = unsafe { perm.as_underlying() }?;
+
+    client.rt.block_on(
+        client
+            .inner
+            .team(team.into())
+            .add_perm_to_role(role.into(), perm),
+    )?;
+    Ok(())
```

Comment:
We could probably map errors so the result could be returned directly:
```suggestion
    client.rt.block_on(
        client
            .inner
            .team(team.into())
            .add_perm_to_role(role.into(), perm),
    ).map_err(...)
```